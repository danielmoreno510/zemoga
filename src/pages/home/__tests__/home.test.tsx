import { fireEvent, render } from "@testing-library/react";
import Home from "../home";

jest.mock("axios", () => {
  return {
    create: jest.fn(() => ({
      get: jest.fn(() => ({ data: [] })),
      interceptors: {
        request: { use: jest.fn(), eject: jest.fn() },
        response: { use: jest.fn(), eject: jest.fn() },
      },
    })),
  };
});

describe("Home component", () => {
  describe("GIVEN a set of valid props", () => {
    const testProps = {
      rulingList: [],
      fetchVote: jest.fn(),
    };

    describe("WHEN the component is rendered", () => {
      const { container } = render(<Home {...testProps} />);

      it("THEN should display a regular Home", () => {
        expect(container).toMatchSnapshot();
      });
    });

    describe("WHEN the upButton and downButton button is pressed", () => {
      const container = render(<Home {...testProps} />);
      const upButton = container.getAllByTestId("upButton");
      const downButton = container.getAllByTestId("downButton");
      fireEvent.click(upButton[0])
      fireEvent.click(downButton[0])

      it('THEN fetchVote function should be called 2 times', () => {
        expect(testProps.fetchVote).toBeCalledTimes(2);
      });
    });
    
  });
});
