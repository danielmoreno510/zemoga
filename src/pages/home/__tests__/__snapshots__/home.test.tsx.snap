// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component GIVEN a set of valid props WHEN the component is rendered THEN should display a regular Home 1`] = `
<div>
  <header
    class="hero"
  >
    <img
      alt="Pope Francis"
      class="hero__background"
      data-nimg="fill"
      decoding="async"
      loading="lazy"
      sizes="(min-width: 750px) 1440px, 100vw"
      src="/_next/image?url=%2Fpope-francis.png&w=3840&q=75"
      srcset="/_next/image?url=%2Fpope-francis.png&w=640&q=75 640w, /_next/image?url=%2Fpope-francis.png&w=750&q=75 750w, /_next/image?url=%2Fpope-francis.png&w=828&q=75 828w, /_next/image?url=%2Fpope-francis.png&w=1080&q=75 1080w, /_next/image?url=%2Fpope-francis.png&w=1200&q=75 1200w, /_next/image?url=%2Fpope-francis.png&w=1920&q=75 1920w, /_next/image?url=%2Fpope-francis.png&w=2048&q=75 2048w, /_next/image?url=%2Fpope-francis.png&w=3840&q=75 3840w"
      style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
    />
    <div
      class="max-centered"
    >
      <div
        class="hero__featured-card"
      >
        <div
          class="featured-card__glass-background"
        />
        <div
          class="featured-card__content"
        >
          <p
            class="featured-card__hairline"
          >
            What's your opinion on
          </p>
          <h2
            class="featured-card__title"
          >
            Pope Francis?
          </h2>
          <p
            class="featured-card__desc"
          >
            He’s talking tough on clergy sexual abuse, or is he just another pervert protector? (thumbs down) or a true pedophile punishing pontiff? (thumbs up)
          </p>
          <p
            class="featured-card__more-info"
          >
            <a
              href="http://wikipedia.com"
            >
              <svg
                class="featured-card__more-info-icon"
                preserveAspectRatio="true"
                viewBox="0 0 27 18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M27 .303c0 .1-.032.2-.09.28a.255.255 0 01-.2.125 2.46 2.46 0 00-1.453.602 5.676 5.676 0 00-1.166 1.952l-6.127 14.533c-.04.135-.152.203-.337.203a.374.374 0 01-.337-.203l-3.436-7.564-3.952 7.564a.374.374 0 01-.337.203.34.34 0 01-.349-.203L3.196 3.262a5.052 5.052 0 00-1.19-1.89A3.161 3.161 0 00.267.708.23.23 0 01.086.6.378.378 0 010 .355C0 .118.064 0 .192 0 .73 0 1.29.025 1.876.075c.544.053 1.056.078 1.536.078.49 0 1.067-.026 1.732-.078C5.839.025 6.456 0 6.994 0c.128 0 .192.118.192.355 0 .235-.04.352-.119.352a2.308 2.308 0 00-1.268.43c-.297.22-.47.581-.463.963.015.263.08.521.192.757l4.975 11.826 2.824-5.614-2.631-5.807A7.637 7.637 0 009.53 1.257a2.274 2.274 0 00-1.382-.55A.208.208 0 017.986.6a.4.4 0 01-.078-.245c0-.237.054-.355.168-.355.494-.002.987.023 1.477.075.46.054.92.08 1.382.078.48 0 .988-.026 1.525-.078C13.013.025 13.558 0 14.094 0c.128 0 .192.118.192.355 0 .235-.038.352-.119.352-1.073.078-1.61.399-1.61.963.047.414.174.814.373 1.175l1.74 3.72 1.732-3.403c.209-.37.333-.786.36-1.215 0-.775-.536-1.188-1.61-1.24-.097 0-.144-.117-.144-.352a.44.44 0 01.071-.24c.05-.077.098-.115.145-.115.385 0 .857.025 1.418.075.536.053.978.078 1.322.078a13.6 13.6 0 001.093-.065A16.806 16.806 0 0120.584 0c.095 0 .142.1.142.303 0 .27-.088.405-.263.405-.54.036-1.061.224-1.508.544a6.937 6.937 0 00-1.423 2.01l-2.308 4.492 3.125 6.702 4.614-11.294c.149-.36.23-.745.24-1.137 0-.828-.537-1.267-1.61-1.317-.097 0-.145-.118-.145-.353 0-.237.071-.355.216-.355.392 0 .857.025 1.394.075.496.053.914.078 1.25.078.409-.003.818-.03 1.224-.078.483-.05.915-.075 1.3-.075.111 0 .168.1.168.303z"
                  fill="#FFF"
                  fill-rule="nonzero"
                />
              </svg>
              More information
            </a>
          </p>
          <p
            class="featured-card__cta"
          >
            What’s Your Veredict?
          </p>
          <div
            class="featured-card__buttons"
          >
            <button
              aria-label="thumbs up"
              class="icon-button home items-center"
              data-testid="upButton"
            >
              <img
                alt="thumbs up"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-up.svg"
                style="color: transparent;"
                width="0"
              />
            </button>
            <button
              aria-label="thumbs down"
              class="icon-button home items-center"
              data-testid="downButton"
            >
              <img
                alt="thumbs down"
                data-nimg="1"
                decoding="async"
                height="0"
                loading="lazy"
                src="/thumbs-down.svg"
                style="color: transparent;"
                width="0"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="hero__closing-gauge"
    >
      <div
        class="closing-gauge__left"
      >
        <span
          class="closing-gauge__title"
        >
          closing in
        </span>
      </div>
      <div
        class="closing-gauge__right"
      >
        <span
          class="closing-gauge__number"
        >
          22
        </span>
        <span
          class="closing-gauge__desc"
        >
          days
        </span>
      </div>
    </div>
  </header>
  <div
    class="max-centered"
  >
    <aside
      aria-label="Speak Out"
      class="banner banner-top"
      role="doc-tip"
    >
      <div
        class="banner__left"
      >
        <span
          class="banner__hairline"
        >
          Speak out. Be heard.
        </span>
        <span
          class="banner__title"
        >
          Be counted
        </span>
      </div>
      <div
        class="banner__right"
      >
        <p
          class="banner__text"
        >
          Rule of Thumb is a crowd sourced court of public opinion where anyone and everyone can speak out and speak freely. It’s easy: You share your opinion, we analyze and put the data in a public report.
        </p>
      </div>
      <button
        aria-label="close"
        class="icon-button"
      >
        <svg
          height="20"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="#000"
            stroke-width="2"
          >
            <path
              d="M1 19L19 1M1 1l18 18"
            />
          </g>
        </svg>
      </button>
    </aside>
    <main
      role="main"
    >
      <div>
        <div
          class="flex justify-end items-baseline"
        >
          <div>
            <button
              class="text-black border-2 border-black border-solid outline-none font-medium text-sm px-5 py-2.5 text-center inline-flex items-center"
              data-dropdown-delay="500"
              data-dropdown-toggle="dropdownDelay"
              data-dropdown-trigger="hover"
              data-testid="open"
              id="dropdownDelayButton"
              type="button"
            >
              List
               
              <svg
                aria-hidden="true"
                class="w-2.5 h-2.5 ml-2.5"
                fill="none"
                viewBox="0 0 10 6"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m1 1 4 4 4-4"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="card-list"
        />
      </div>
    </main>
    <aside
      aria-label="Submit a name"
      class="banner banner-bottom"
      role="doc-tip"
    >
      <img
        alt=""
        class="banner__background"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        role="none"
        sizes="(min-width: 750px) 1440px, 100vw"
        src="/_next/image?url=%2Fbg-people.png&w=3840&q=75"
        srcset="/_next/image?url=%2Fbg-people.png&w=640&q=75 640w, /_next/image?url=%2Fbg-people.png&w=750&q=75 750w, /_next/image?url=%2Fbg-people.png&w=828&q=75 828w, /_next/image?url=%2Fbg-people.png&w=1080&q=75 1080w, /_next/image?url=%2Fbg-people.png&w=1200&q=75 1200w, /_next/image?url=%2Fbg-people.png&w=1920&q=75 1920w, /_next/image?url=%2Fbg-people.png&w=2048&q=75 2048w, /_next/image?url=%2Fbg-people.png&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
      />
      <div
        class="banner__left"
      >
        <h2
          class="banner__heading"
        >
          Is there anyone else you would want us to add?
        </h2>
      </div>
      <div
        class="banner__right"
      >
        <button
          class="banner__cta"
        >
          Submit a name
        </button>
      </div>
    </aside>
    <hr
      role="separator"
    />
  </div>
</div>
`;
